bgen does not support top level namespace declarations

Attempting to process an API definition file with this contents:

```
namespace iosbinding;
```

fails with:

```
ApiDefinition.cs(1,21): error CS1514: { expected [/Users/donblas/Programming/xamarin-macios/tests/dotnet/UnitTests/bin/Debug/net6.0/tmp-test-dir/Xamarin.Tests.TemplateTest.CreateAndBuildTemplate1/iosbinding/iosbinding.csproj]
ApiDefinition.cs(1,22): error CS1513: } expected [/Users/donblas/Programming/xamarin-macios/tests/dotnet/UnitTests/bin/Debug/net6.0/tmp-test-dir/Xamarin.Tests.TemplateTest.CreateAndBuildTemplate1/iosbinding/iosbinding.csproj]
```

But this works:

```
namespace iosbinding {}
```

This is relevant to NET6 as the templates being written for https://github.com/xamarin/xamarin-macios/issues/13578 can't match the namespace general style of the other templates.

--- a/tests/generator/BGenTests.cs
+++ b/tests/generator/BGenTests.cs
@@ -879,6 +879,15 @@ namespace GeneratorTests
 		}
 
 		[Test]
+#if !NET
+		[Ignore ("This only applies to .NET")]
+#endif
+		public void CSharp10Syntax ()
+		{
+			BuildFile (Profile.iOS, "tests/csharp10syntax.cs");
+		}
+
+		[Test]
 		public void NFloatType ()
 		{
 			var bgen = BuildFile (Profile.iOS, "tests/nfloat.cs");

--- a/tests/generator/tests/csharp10syntax.cs
+++ b/tests/generator/tests/csharp10syntax.cs
@@ -1,0 +1,6 @@
+global using Foundation;
+
+namespace Root;
+
+[BaseType (typeof (NSObject))]
+interface IFace {}

